packages/synapse-react-client/src/components/QueryVisualizationWrapper.tsx

```diff
@@ -138,8 +139,12 @@ export function QueryVisualizationWrapper(
   } = useQueryContext()
 
   let { rowSelectionPrimaryKey } = props
-  if (!rowSelectionPrimaryKey && isFileViewOrDataset(entity)) {
-    // If the primary key isn't specified on a file view/dataset, we can safely use the 'id' column
+  if (
+    !rowSelectionPrimaryKey &&
+    isFileViewOrDataset(entity) &&
+    data?.columnModels?.find(cm => cm.name === 'id')
```

Comment:
make sure the "id" column is present before assuming it's the primary key of a file view